paddle/pir/dialect/shape/ir/shape_op.cc

```diff
@@ -226,6 +244,33 @@ pir::Block *FuncOp::block() {
   return region.front();
 }
 
+void TensorDimOp::Build(Builder &builder,
+                        OperationArgument &argument,
+                        const pir::OpResult &source,
+                        const pir::OpResult &index) {
+  argument.inputs = {source, index};
+  argument.output_types.emplace_back(
+      pir::IndexType::get(pir::IrContext::Instance()));
+}
+
+void TensorDimOp::Build(Builder &builder,
+                        OperationArgument &argument,
+                        const pir::OpResult &source,
+                        int64_t index) {
+  pir::OpResult indexValue =
+      builder
+          .Build<pir::ConstantOp>(
+              pir::Int64Attribute::get(pir::IrContext::Instance(), 2),
+              pir::IndexType::get(pir::IrContext::Instance()))
+          ->result(0);
+  argument.inputs = {source, indexValue};
+  argument.output_types.emplace_back(
+      pir::IndexType::get(pir::IrContext::Instance()));
+}
+
+pir::Value TensorDimOp::getSource() { return operand_source(0); }
```

Comment:
pd_op.h 因为是与yaml对应的，所以在暴露输入输出接口时，用的是 XXOp.x()、XXOp.bias() 这样的形态，这里可以看下是否要对齐为：
```suggestion
pir::Value TensorDimOp::source() { return operand_source(0); }
```

Reply:
可以对齐，将在下个 PR 中进行调整。